pkg/core/native/policy.go

```diff
@@ -0,0 +1,353 @@
+	defaultMaxTransactionsPerBlock = 512
+	defaultFeePerByte              = 1000
+)
+
+var (
+	maxTransactionsPerBlockKey = []byte{23}
+	feePerByteKey              = []byte{10}
+	blockedAccountsKey         = []byte{15}
+	maxBlockSizeKey            = []byte{16}
+)
+
+// Policy represents Policy native contract.
+type Policy struct {
+	interop.ContractMD
+}
+
+var _ interop.Contract = (*Policy)(nil)
+
+// newPolicy returns Policy native contract.
+func newPolicy() *Policy {
+	p := &Policy{ContractMD: *interop.NewContractMD(policySyscallName)}
+
+	p.ContractID = policyContractID
+	p.Manifest.Features |= smartcontract.HasStorage
+
+	desc := newDescriptor("getMaxTransactionsPerBlock", smartcontract.IntegerType)
+	md := newMethodAndPrice(p.getMaxTransactionsPerBlock, 1, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, true)
+
+	desc = newDescriptor("getMaxBlockSize", smartcontract.IntegerType)
+	md = newMethodAndPrice(p.getMaxBlockSize, 1, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, true)
+
+	desc = newDescriptor("getFeePerByte", smartcontract.IntegerType)
+	md = newMethodAndPrice(p.getFeePerByte, 1, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, true)
+
+	desc = newDescriptor("getBlockedAccounts", smartcontract.ArrayType)
+	md = newMethodAndPrice(p.getBlockedAccounts, 1, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, true)
+
+	desc = newDescriptor("setMaxBlockSize", smartcontract.BoolType,
+		manifest.NewParameter("value", smartcontract.IntegerType))
+	md = newMethodAndPrice(p.setMaxBlockSize, 3, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("setMaxTransactionsPerBlock", smartcontract.BoolType,
+		manifest.NewParameter("value", smartcontract.IntegerType))
+	md = newMethodAndPrice(p.setMaxTransactionsPerBlock, 3, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("setFeePerByte", smartcontract.BoolType,
+		manifest.NewParameter("value", smartcontract.IntegerType))
+	md = newMethodAndPrice(p.setFeePerByte, 3, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("blockAccount", smartcontract.BoolType,
+		manifest.NewParameter("account", smartcontract.Hash160Type))
+	md = newMethodAndPrice(p.blockAccount, 3, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("unblockAccount", smartcontract.BoolType,
+		manifest.NewParameter("account", smartcontract.Hash160Type))
+	md = newMethodAndPrice(p.unblockAccount, 3, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+	return p
+}
+
+// Metadata implements Contract interface.
+func (p *Policy) Metadata() *interop.ContractMD {
+	return &p.ContractMD
+}
+
+// Initialize initializes Policy native contract and implements Contract interface.
+func (p *Policy) Initialize(ic *interop.Context) error {
+	if ic.Trigger != trigger.Application {
+		return errors.Errorf("invalid trigger type: %v", ic.Trigger.String())
+	}
+
+	if err := ic.DAO.PutStorageItem(p.Hash, maxBlockSizeKey, &state.StorageItem{
```

Comment:
Please, let's keep `if` condition in one line, it is much harder to read when it is not so.
`err := ic.DAO...` of `si := &state.StorageItem{...`can be done on another line.

Reply:
And I prefer not mixing assignments and conditions at all, like:
```
err := ...
if err != nil
```
One additional line makes it so much easier.